plugin/src/main/java/org/gradle/testretry/internal/config/TestTaskConfigurer.java

```diff
@@ -113,6 +114,17 @@ private static boolean supportsFinalizeValueOnRead(GradleVersion gradleVersion)
         return gradleVersion.compareTo(GRADLE_6_1) >= 0;
     }
 
+    private static boolean supportsDryRun(GradleVersion gradleVersion) {
+        return gradleVersion.compareTo(GRADLE_8_3) >= 0;
+    }
+
+    private static boolean callGetDryRun(GradleVersion gradleVersion, Test task) {
+        if (supportsDryRun(gradleVersion)) {
+            return invoke(declaredMethod(Test.class, "getDryRun"), task);
```

Comment:
❌ Rather than using reflection, we should update the build to a Gradle nightly version and call it directly (behind the same if).